### PR TITLE
Repair - Make Full Repair Time dependent on vehicle damage

### PR DESCRIPTION
**Pull request by LinkIsGrim**
**When merged this pull request will:**
- Title.
- Add getHitPointsToIgnore function and move functionality from fnc_addRepairActions.
- Modify addRepairActions to use getHitPointsToIgnore.
- Add CBA Settings for configurable time for Misc and Wheel repairs.
- Refactor fnc_isInRepairFacility and fnc_isNearRepairVehicle to use cached call.
- Add `forceDisplayName` property to `ACE_Repair >> Actions` for repair actions that don't want to show the hitpoint name.
- Add `loopAnimation` property to `ACE_Repair >> Actions` to loop the action's animation.
- Change some stringtables. Namely `Wheel Repair` > `Wheel Change`, now that Wheel Patching is a thing.
- Add Repair Time Coefficient setting. Modifies time for individual repairs while doing full repair (and by consequence, total time).

Full Repair is now incremental and time is based on amount of time needed to do individual repairs + another repair for inacessible hitpoints, if applicable.

### CODE REVIEW

**Comment by Salluci** [comment]
Works for me. Something like surgicalKitProgress logic, then?

**Comment by veteran29** [comment]
> Works for me. Something like surgicalKitProgress logic, then?

Yeah, I guess so.

**Comment by Salluci** [comment]
I'd appreciate some help or pointers on incorporating looping animations or something of the sort into this.

**Comment by jonpas** [comment]
Pushing this to next release as it seems it needs some help with animation stuff and hasn't been reviewed properly yet.

**Comment by LinkIsGrim** [comment]
Still needs animation/sound work, but functionality is done.

**Comment by LinkIsGrim** [comment]
Animations loop (though I'm not a fan). Ready for review.

File: addons/repair/functions/fnc_repair.sqf [hunk 1]
```diff
@@ -171,10 +172,27 @@ if (vehicle _caller == _caller && {_callerAnim != ""}) then {
...
 };
 
+if (_loopAnim) then {
+    private _animDoneEh = _caller addEventHandler ["AnimDone", {
+        params ["_caller", "_anim"];
+        if (_anim isEqualTo (_caller getVariable [QGVAR(repairCurrentAnimCaller), ""])) then {
+            [{
+                params ["_caller", "_anim"];
+                if !(isNil {_caller getVariable QGVAR(repairCurrentAnimCaller)}) then {
+                    systemChat "loop";
```
Comment (veteran29) [inline addons/repair/functions/fnc_repair.sqf:187]:
```suggestion
                    TRACE_1("loop",_caller);
```

**Review by veteran29 — APPROVED** [review]
LGTM

**Pull request merged**